addons/stock/models/product.py

```diff
@@ -576,6 +576,8 @@ def write(self, vals):
             ])
             if existing_move_lines:
                 raise UserError(_("You can not change the type of a product that is currently reserved on a stock move. If you need to change the type, you should first unreserve the stock move."))
+        if 'type' in vals and vals['type'] != 'product' and self.qty_available != 0:
```

Comment:
`self.qty_available != 0` is not good, it is better to use `float_zero(self.qty_available, precision_rouding=self.uom_id.rouding)`

Reply:
thanks for the suggestion